lib/ansible/modules/service_facts.py

```diff
@@ -45,6 +45,18 @@
 - name: Print service facts
   ansible.builtin.debug:
     var: ansible_facts.services
+
+- name: show only existing systemd services
+  debug: msg={{existing_systemd_services}}
+  vars:
+     existing_systemd_services: "{{ ansible_facts['services'] | selectattr('status', 'defined') | rejectattr('status', 'equalto', 'not-found') }}"
```

Comment:
```suggestion
     existing_systemd_services: "{{ ansible_facts['services'].values() | selectattr('status', 'defined') | rejectattr('status', 'equalto', 'not-found') }}"
```
Need to iterate over the values, not the keys. Could also use `dict2items`, but the values include a `name` attribute so I'm not sure that gains you anything.